### PR TITLE
Replace gh-pages branch with Jekyll workflow.

### PR DESCRIPTION
**Pull request by Endermanbugzjfc**
#### README preview
https://zjfc.is-a.dev/await-generator/
#### Book preview
https://zjfc.is-a.dev/await-generator/master/
# Please enable this before merging!
image

### CODE REVIEW

File: .github/workflows/book.yml [hunk 1]
```diff
@@ -20,9 +33,26 @@ jobs:
...
-      - run: git add $(echo ${{github.ref}} | cut -d/ -f3) && git commit --allow-empty -m "Docs build for SOF3/await-generator@${{github.sha}}"
-      - run: git push
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./
+          destination: ./_site
```
Comment (SOF3) [inline .github/workflows/book.yml:45]: Does this mean we cannot persist pages on other branches/tags?

File: .github/workflows/book.yml [hunk 2]
```diff
@@ -20,9 +33,26 @@ jobs:
...
       - run: test ! -d $(echo ${{github.ref}} | cut -d/ -f3) || rm -r $(echo ${{github.ref}} | cut -d/ -f3)
       - run: cp -r ../book $(echo ${{github.ref}} | cut -d/ -f3)
-      - run: git config --local user.name "github-actions[bot]" && git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
-      - run: git add $(echo ${{github.ref}} | cut -d/ -f3) && git commit --allow-empty -m "Docs build for SOF3/await-generator@${{github.sha}}"
-      - run: git push
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
```
Comment (SOF3) [inline .github/workflows/book.yml:42]: The book output is already HTML, why do we need another jekyll workflow here?

**Review by SOF3 — CHANGES_REQUESTED** [review]

File: .github/workflows/book.yml (hunk 2 above)
Comment (Endermanbugzjfc) [reply .github/workflows/book.yml:42]: To build the README.

File: .github/workflows/book.yml (hunk 1 above)
Comment (Endermanbugzjfc) [reply .github/workflows/book.yml:45]: Let me add a download artifact action.

Comment (SOF3) [reply .github/workflows/book.yml:45]: @Endermanbugzjfc artifact of what? previous action run?

Comment (Endermanbugzjfc) [reply .github/workflows/book.yml:45]: > What is the point of spending so much effort to manually manage the previous artifact versions when the current workflow already works, using Git VCS to help us manage multiple versions?

For faster clone speed.

Comment (Endermanbugzjfc) [reply .github/workflows/book.yml:45]: Well, there is still a problem tho. The current [README](https://sof3.github.io/await-generator/) is not up to date.

**Comment by SOF3** [comment]
By the way, if cloning the `gh-pages` diff is too slow, we could make it force push every time.

**Pull request closed without merging**